src/raw_client.rs

```diff
@@ -681,32 +681,27 @@ impl<T: Read + Write> ElectrumApi for RawClient<T> {
 
         self.increment_calls();
 
-        while !missing_responses.is_empty() {
-            let req_id = *missing_responses.iter().next().unwrap();
-            let resp = match self.recv(&receiver, req_id) {
-                Ok(resp) => resp,
+        for req_id in missing_responses.iter() {
+            match self.recv(&receiver, *req_id) {
+                Ok(mut resp) => answers.insert(
+                    resp["id"].as_u64().unwrap_or_default(),
```

Comment:
I'm not entirely sure `unwrap_or_default()` is a good idea here: in theory if we got to this point it means that the response had an `id`, but in case it's missing assuming its value is `0` can create issues. I think it would be better to replace it with `.unwrap_or(req_id)`, or even just always returning the `req_id` directly.

Reply:
You are right. Fixed.